src/Device/DeviceType.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+
+interface DeviceType {
+  // Specified name for certain device type.
+  name(): string;
+
+  // device Executable ext
+  getExecutableExt(): string[];
+
+  // list specified name for connectable devices
+  getConnectableDevices(): Command;
+
+  // check certain device connected
+  checkConnected(name: string): Command;
```

Comment:
Little bit confused as this will return real result(like `string[]` or `boolean` on this.